homeassistant/components/tuya/fan.py

```diff
@@ -110,34 +113,33 @@ def __init__(self, device: TuyaDevice, device_manager: TuyaDeviceManager) -> Non
         """Init Tuya Fan Device."""
         super().__init__(device, device_manager)
 
+        self.ha_preset_modes = []
+        if DPCODE_MODE in self.tuya_device.function:
+            self.ha_preset_modes = json.loads(
+                self.tuya_device.function.get(DPCODE_MODE, {}).values
```

Comment:
Same problem here as in the other PRs. `json.loads` will error if we pass it `dict.values`.